### DISCUSSION TITLE
how to output the log in server action

### DISCUSSION THREAD

**Discussion by sytpb** [vercel/next.js#52181] [Help] [answered]
### Summary

how output the log in server action  

````
async function action() {
        console.log("submit....")


        alert("yyyyy")
    }

    return (
        <>
                <form action={action} className="container mx-auto w-6/12">
<button type='submit' className="btn bg-primary btn-wide text-white hover:bg-primary mx-auto flex m-2">下一步</button>
.....
````

  console.log not work this time .

**Comment by icyJoseph** [comment]
It logs on the server. Server Actions are invoked from the client, but run server side.

**Reply by sytpb** [reply]
thank for you reply ,  Now I run the code on local(npm run dev), where can i get the console log ?

**Reply by icyJoseph** [reply]
In the terminal where you run that command

**Reply by sytpb** [reply]
![image](https://github.com/vercel/next.js/assets/12178686/275fcd18-06b0-4086-97a6-c5274ec5f32d)

**Reply by icyJoseph** [reply]
Did you remove the `alert` call too? I think that might cause errors.

Also your function is missing the `'use server'` directive:

```ts
     async function action() {
        "use server"
        console.log("submit....")
    }
```

And also, activate server actions on your `next.config.js`

**Reply by sytpb** [reply, marked as answer]
![image](https://github.com/vercel/next.js/assets/12178686/be9f35d2-f596-4161-abee-4a7ac98b42a2)


![image](https://github.com/vercel/next.js/assets/12178686/0772c85f-098e-4b19-891d-4a105b1b0963)

OK ,you are right , I have understood , I should put the console.log in action file ,   not in client component . 

really appriciate  for you .

**Reply by icyJoseph** [reply]
But if, what you send into the form is the `action` function, then that also needs to be marked with `use server`, AFAIK anyway.

**Reply by sytpb** [reply]
no need 'use server ' in client component .
